osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/ScanMbusChannelsCommandExecutor.java

```diff
@@ -28,77 +28,49 @@
+  /** IDs of the attributes of the M-Bus Client Setup that make up the Short ID. */
+  private static final MbusClientAttribute[] ATTRIBUTE_IDS_SHORT_ID =
+      new MbusClientAttribute[] {
+        MbusClientAttribute.IDENTIFICATION_NUMBER,
+        MbusClientAttribute.MANUFACTURER_ID,
+        MbusClientAttribute.VERSION,
+        MbusClientAttribute.DEVICE_TYPE
       };
 
-  private static final int OBIS_BYTE_A_MBUS_CLIENT_SETUP = 0;
-  private static final int OBIS_BYTE_C_MBUS_CLIENT_SETUP = 24;
-  private static final int OBIS_BYTE_D_MBUS_CLIENT_SETUP = 1;
-  private static final int OBIS_BYTE_E_MBUS_CLIENT_SETUP = 0;
-  private static final int OBIS_BYTE_F_MBUS_CLIENT_SETUP = 255;
-
   private static final int NUMBER_OF_CHANNELS = 4;
   private static final int NUMBER_OF_SHORT_ID_ATTRIBUTES_PER_CHANNEL =
       ATTRIBUTE_IDS_SHORT_ID.length;
 
-  private static final AttributeAddress[] SHORT_ID_ATTRIBUTE_ADDRESSES =
-      makeAttributeAddressesShortIds();
-
-  @Autowired private DlmsHelper dlmsHelper;
-
-  @Autowired private DlmsObjectConfigService dlmsObjectConfigService;
+  private final DlmsHelper dlmsHelper;
+  private final ObjectConfigServiceHelper objectConfigServiceHelper;
 
-  public ScanMbusChannelsCommandExecutor() {
+  public ScanMbusChannelsCommandExecutor(
+      final DlmsHelper dlmsHelper, final ObjectConfigServiceHelper objectConfigServiceHelper) {
     super(ScanMbusChannelsRequestDataDto.class);
+    this.dlmsHelper = dlmsHelper;
+    this.objectConfigServiceHelper = objectConfigServiceHelper;
   }
 
-  private static ObisCode getObisCodeMbusClientSetup(final int channel) {
-    return new ObisCode(
-        OBIS_BYTE_A_MBUS_CLIENT_SETUP,
-        channel,
-        OBIS_BYTE_C_MBUS_CLIENT_SETUP,
-        OBIS_BYTE_D_MBUS_CLIENT_SETUP,
-        OBIS_BYTE_E_MBUS_CLIENT_SETUP,
-        OBIS_BYTE_F_MBUS_CLIENT_SETUP);
-  }
-
-  /**
-   * @see #ATTRIBUTE_IDS_SHORT_ID
-   * @see #channelShortIdsFromGetResults(List, Protocol)
-   */
-  private static AttributeAddress[] makeAttributeAddressesShortIds() {
+  private AttributeAddress[] makeAttributeAddressesShortIds(
+      final DlmsConnectionManager conn, final DlmsDevice device) throws ProtocolAdapterException {
     final AttributeAddress[] shortIdAddresses =
         new AttributeAddress[NUMBER_OF_CHANNELS * NUMBER_OF_SHORT_ID_ATTRIBUTES_PER_CHANNEL];
     int index = 0;
     for (int channel = 1; channel <= NUMBER_OF_CHANNELS; channel++) {
-      final ObisCode obisCode = getObisCodeMbusClientSetup(channel);
+
+      final CosemObjectAccessor cosemObjectAccessor =
+          this.createCosemObjectAccessor(conn, device, (short) channel);
+
       for (int i = 0; i < NUMBER_OF_SHORT_ID_ATTRIBUTES_PER_CHANNEL; i++) {
         shortIdAddresses[index++] =
-            new AttributeAddress(CLASS_ID, obisCode, ATTRIBUTE_IDS_SHORT_ID[i]);
+            cosemObjectAccessor.createAttributeAddress(ATTRIBUTE_IDS_SHORT_ID[i]);
```

Comment:
Why do we need the CosemObjectAccessor? I think it's easier to get the CosemObject for MBUS_CLIENT_SETUP from the object config and then create the AttributeAddresses. For example, the create the CosemObjectAccessor you need to pass the connection, but that connection is never used. The cosemObject can then also be reused to get the version.

Reply:
The commandexecutors working with channel dependent cosem objects are now using ObjectConfigService(Helper), CosemObjectAccessor and DeviceChannelHelper classes. Responsibilities are not clear and overlapping. Some redefining and refactoring is needed but out of scope for this PR.